presto-orc/src/main/java/com/facebook/presto/orc/stream/MissingInputStreamSource.java

```diff
@@ -15,19 +15,91 @@
+    private static final MissingInputStreamSource<LongInputStream> longMissingStreamSource = new MissingInputStreamSource<>(LongInputStream.class);
+    private static final MissingInputStreamSource<RowGroupDictionaryLengthInputStream> rowGroupDictionaryLengthMissingStreamSource = new MissingInputStreamSource<>(RowGroupDictionaryLengthInputStream.class);
+
     private final Class<S> streamType;
 
-    public static <S extends ValueInputStream<?>> InputStreamSource<S> missingStreamSource(Class<S> streamType)
+    private MissingInputStreamSource(Class<S> streamType)
     {
-        return new MissingInputStreamSource<>(streamType);
+        this.streamType = requireNonNull(streamType, "streamType is null");
     }
 
-    private MissingInputStreamSource(Class<S> streamType)
+    public static InputStreamSource<BooleanInputStream> getBooleanMissingStreamSource()
+    {
+        return booleanMissingStreamSource;
+    }
+
+    public static InputStreamSource<ByteInputStream> getByteMissingStreamSource()
+    {
+        return byteMissingStreamSource;
+    }
+
+    public static MissingInputStreamSource<ByteArrayInputStream> getByteArrayMissingStreamSource()
+    {
+        return byteArrayMissingStreamSource;
+    }
+
+    public static MissingInputStreamSource<DecimalInputStream> getDecimalMissingStreamSource()
+    {
+        return decimalMissingStreamSource;
+    }
+
+    public static MissingInputStreamSource<DoubleInputStream> getDoubleMissingStreamSource()
+    {
+        return doubleMissingStreamSource;
+    }
+
+    public static MissingInputStreamSource<FloatInputStream> getFloatMissingStreamSource()
+    {
+        return floatMissingStreamSource;
+    }
+
+    public static InputStreamSource<LongInputStream> getLongMissingStreamSource()
     {
-        this.streamType = streamType;
+        return longMissingStreamSource;
+    }
+
+    public static MissingInputStreamSource<RowGroupDictionaryLengthInputStream> getRowGroupDictionaryLengthMissingStreamSource()
+    {
+        return rowGroupDictionaryLengthMissingStreamSource;
+    }
+
+    public static <S extends ValueInputStream<?>> InputStreamSource<S> missingStreamSource(Class<S> streamType)
+    {
+        if (booleanMissingStreamSource.streamType.equals(streamType)) {
+            return (InputStreamSource<S>) booleanMissingStreamSource;
+        }
+        if (byteMissingStreamSource.streamType.equals(streamType)) {
+            return (InputStreamSource<S>) byteMissingStreamSource;
+        }
+        if (byteArrayMissingStreamSource.streamType.equals(streamType)) {
+            return (InputStreamSource<S>) byteArrayMissingStreamSource;
+        }
+        if (decimalMissingStreamSource.streamType.equals(streamType)) {
+            return (InputStreamSource<S>) decimalMissingStreamSource;
+        }
+        if (floatMissingStreamSource.streamType.equals(streamType)) {
```

Comment:
doubleMissingStreamSource is missing

Reply:
Wow! great catch, among a list of repetitive changes.